pkg/venafi/firefly/connector.go

```diff
@@ -114,6 +114,11 @@ func (c *Connector) Authorize(auth *endpoint.Authentication) (token *oauth2.Toke
 		return config.PasswordCredentialsToken(context.Background(), auth.User, auth.Password)
 	}
 
+	// if it's a device flow grant
+	if auth.IdentityProvider.DeviceURL != "" {
+		return c.getDeviceAccessToken(auth)
```

Comment:
You can seperate the function from here and verify err, if not error then print log success log and return values:
```
if auth.IdentityProvider.DeviceURL != "" {
                authToken, err = c.getDeviceAccessToken(auth)
                if err != nil {
                   return nil, err
                }
                log.Printf("Successfully authorized device) // Not sure about the log syntax
		return authToken, nil
}
```

Reply:
It's done now into the c.getDeviceAccessToken(auth)